haystack/nodes/prompt/invocation_layer/hugging_face_inference.py

```diff
@@ -102,49 +123,78 @@ def invoke(self, *args, **kwargs):
                 f"No prompt provided. Model {self.model_name_or_path} requires prompt."
                 f"Make sure to provide prompt in kwargs."
             )
-        stop_words = kwargs.pop("stop_words", None)
-
+        prompt = self.preprocess_prompt(prompt)
+        stop_words = kwargs.pop("stop_words", None) or []
         kwargs_with_defaults = self.model_input_kwargs
-        if "max_new_tokens" not in kwargs_with_defaults:
-            kwargs_with_defaults["max_new_tokens"] = self.max_length
+        if kwargs:
+            if "max_new_tokens" not in kwargs_with_defaults:
+                kwargs_with_defaults["max_new_tokens"] = self.max_length
+            kwargs_with_defaults.update(kwargs)
+
+        # either stream is True (will use default handler) or stream_handler is provided
+        stream = (
+            kwargs_with_defaults.get("stream", False) or kwargs_with_defaults.get("stream_handler", None) is not None
+        )
 
-        if "top_k" in kwargs:
-            top_k = kwargs.pop("top_k")
-            kwargs["num_return_sequences"] = top_k
         kwargs_with_defaults.update(kwargs)
```

Comment:
Let's remove this update as we're doing it already above and we don't change `kwargs_with_default` further after that update.